xarray/backends/scipy_.py

```diff
@@ -37,30 +37,39 @@ def __init__(self, variable_name, datastore):
 
     @property
     def array(self):
-        return self.datastore.ds.variables[self.variable_name].data
+        with self.datastore.ensure_open(autoclose=True):
+            return self.datastore.ds.variables[self.variable_name].data
 
     @property
     def dtype(self):
-        # always use native endianness
-        return np.dtype(self.array.dtype.kind + str(self.array.dtype.itemsize))
+        with self.datastore.ensure_open(autoclose=True):
```

Comment:
We want to use the same approach for scipy as for netcdf4 to avoid repeatedly opening/closing files (i.e., we should cache dtype). Each time we open, scipy does a bunch of logic in python to read in the metadata for the entire netCDF file.